plugins/test/dynamic_test.cc

```diff
@@ -280,19 +290,35 @@ void DynamicTest::BenchHttpHandlers(benchmark::State& state) {
     BM_RETURN_IF_ERROR(headers.status());
     response_headers = *headers;
   }
+  std::vector<std::string> request_body_chunks;
+  for (const auto& request_body : cfg_.request_body()) {
+    request_body_chunks.emplace_back(request_body.input().content());
+  }
+  std::vector<std::string> response_body_chunks;
+  for (const auto& response_body : cfg_.response_body()) {
+    response_body_chunks.emplace_back(response_body.input().content());
+  }
   for (auto _ : state) {
     if (request_headers) {
       auto res = stream.SendRequestHeaders(*request_headers);
       benchmark::DoNotOptimize(res);
       BM_RETURN_IF_FAILED(handle);
     }
-    // Future: send request BODY here.
+    for (std::string& body : request_body_chunks) {
+      auto res = stream.SendRequestBody(body);
```

Comment:
Note that `SendRequestBody(body)` copies `body`. This copy is unavoidable, since there's only one set of `request_body_chunks` vs. many iterations of the outer `for (auto _ : state)` loop, so can't `std::move` out of `request_body_chunks`.

However, if you want to make sure that the string copy is not included in benchmark numbers, you could explicitly copy the request and response body strings at the top of the loop, inside of a `PauseTiming()` + `ResumeTiming()` block (which you might need to add anyways around creation of the `TestHttpContext`, per earlier comment). IOW, you could do:

```
for (auto _ : state) {
  state.PauseTiming();
  auto stream = TestHttpContext(handle);
  std::vector<std::string> request_body_chunk_copies = request_body_chunks;
  std::vector<std::string> response_body_chunk_copies = response_body_chunks;
  state.ResumeTiming();

  // ...

  for (std::string& body : response_body_chunk_copies) {
    auto res = stream.SendResponseBody(std::move(body));
    benchmark::DoNotOptimize(res);
    BM_RETURN_IF_FAILED(handle);
  }
  // ...
}
  }
}
```